lib/eco/treelexer/lexer.py

```diff
@@ -83,6 +84,17 @@ def match_or(self, pattern, text):
         self.pos = tmp # backtrack
         return self.match(pattern.rhs, text)
 
+    def match_range(self, pattern, text):
+        if self.pos >= len(text):
+            return False
+
+        s = ord(pattern.c[0])
+        e = ord(pattern.c[2])
```

Comment:
Do you check that `s` is less than `e` somewhere? If not, add an `assert` here (otherwise you're going to get weirdness if you do `[z-a]`.

Reply:
No here, but one of the later commits checks for this.